### DISCUSSION TITLE
.npmrc requirement

### DISCUSSION THREAD

**Discussion by mark-ship-it** [npm/feedback#140] [CLI] [answered]
In order to work with a private repo, it seems that you always have to run `npm login` or create an .npmrc file with this in it:

```
//registry.npmjs.org/:_authToken=MY_NPM_TOKEN
```

the issue is described more here: https://docs.npmjs.com/using-private-packages-in-a-ci-cd-workflow

It would be helpful if `npm` used the `NPM_TOKEN` environment variable and not require an .npmrc file.  this would save lots of unnecessary files from getting created.

**Comment by ahammond** [comment]
Same problem here. The .npmrc file is particularly obnoxious in CI/CD environments. Having the credential in the environment is how many other similar tools do it and strikes me as the right way to go.

**Comment by isaacs** [comment, marked as answer]
We could make it respect a top-level config setting pretty easily.  We'd just need to be careful not to leak it to install scripts by default, which is a little bit of custom logic for this one config in https://github.com/npm/config, but not a ton.

If we go that route, the path of least effort would be to make the env var `npm_config_token=MY_NPM_TOKEN npm do stuff` or `npm --token=MY_NPM_TOKEN do stuff`.  Would that meet the need here?
